usr/bin/pop-zram-config

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# Copyright 2022 System76 <info@system76.com>
+# SPDX-License-Identifier: GPL-3.0-only
+
+# If a zram device already exists, ignore
+test -b /dev/zram0 && exit 0
+
+ENV_CONF='/etc/pop-zram'
+
+# Load user-defined variabls if a config exists
+if test -f ${ENV_CONF}; then
+    export $(grep -v '^#' ${ENV_CONF} | xargs)
+fi
+
+# Default to compress with zstd, which has an average compression ratio of 3.37.
+test -n "${ALGO}" || ALGO=zstd
+
+# Max amount of uncompressed RAM that should be compressed.
+test -n "${MAX_SIZE}" || MAX_SIZE=16384
+
+# Higher values encourage the kernel to be more eager to move pages to swap.
+test -n "${SWAPPINESS}" || SWAPPINESS=180
+
+# Amount of memory to use for zram, from 1 to 200
+test -n "${PORTION}" || PORTION=100
+
+if ((PORTION > 200)); then
+    PORTION=200
+elif ((PORTION < 1)); then
+    PORTION=1
+fi
+
+# Number of consecutive pages to read in advance. Higher values increase compression
+# for zram devices, but at the cost of significantly reduced IOPS and higher latency.
+# It is highly recommended to use 0 for zstd; and 1 for speedier algorithms.
+test -n "${PAGE_CLUSTERS}" || PAGE_CLUSTERS=$(test zstd = ${ALGO} && echo 0 || echo 1)
+
+# Total amount of memory accessible to the OS.
+test -n "${PAGE_CLUSTERS}" || PAGE_CLUSTERS=$(test zstd = ${ALGO} && echo 0 || echo 1)
+
+# Total amount of memory accessible to the OS.
+TOTAL=$(awk -v p=${PORTION} '/MemTotal/ {printf "%.0f", p * $2 / 102400}' /proc/meminfo)
+
+# Ensure that the size is not greater than the max size.
+SIZE=$(((TOTAL > MAX_SIZE)) && echo ${MAX_SIZE} || echo ${TOTAL})
+
+# Load the zram module.
+modprobe zram num_devices=1 && sync
+sleep 1
+
+# Create it with our desired size and compression algorithm
+zramctl --size "${SIZE}M" --algorithm "${ALGO}" /dev/zram0
+
+# Format it as a swap partition
+mkswap /dev/zram0 && sync
+sleep 1
```

Comment:
I may be a bit paranoid, but sleeps like this concern me. Why the one seconds sleeps here? As well, the 1% of free memory seem a bit arbitrary. Is there maybe a better way to do  this?

Reply:
Because the commands exit before the kernel's finished initializing them. The zram device may be created and the kernel may expose it, but performing mkfs on it will fail with `Device or resource busy`. And likewise the device may have just been formatted as swap, but may fail to enable with swapon with `Device or resource busy`. Typically a script will use `udevadm settle` and sleep for 1 second because the process takes around 0.01 seconds.

Reply:
The recommended setting for `min_free_kbytes` is between 0.5% and 5% of total memory. The current setting is closer to 0.5% and typically fails to trigger OOM before the system freezes. I've been seeing a lot of recommendations for 1%. Some may want a higher value, so if it's set higher than 1% then it won't change the value.

Reply:
> Typically a script will use `udevadm settle` and sleep for 1 second in case that didn't work

I would rather see udevadm used with a timeout (`-t`) than an arbitrary sleep.

Reply:
Added udevadm settle calls

Reply:
If using udevadm then the sleep isn't required and should be removed.